pandas/core/indexes/multi.py

```diff
@@ -1783,16 +1792,15 @@ def is_lexsorted(self):
     @cache_readonly
     def lexsort_depth(self):
         if self.sortorder is not None:
-            if self.sortorder == 0:
-                return self.nlevels
-            else:
-                return 0
+            return int(self.sortorder)
```

Comment:
Is there a reason for `int(...)` here?

Reply:
I took the line from the constructor, line 278, but if it is already cast in the constructor (maybe to handle passing False for sortorder), it may no be necessary here. I look if it passes the test without it.